jdaviz/core/template_mixin.py

```diff
@@ -1979,6 +1987,217 @@ def _check_non_finite_uncertainty_mismatch(self, event={}):
+        super().__init__(*args, **kwargs)
+        self.continuum = SubsetSelect(self,
+                                      'continuum_subset_items',
+                                      'continuum_subset_selected',
+                                      manual_options=['None', 'Surrounding'],
+                                      default_mode='first',
+                                      filters=['is_spectral'])
+
+    def _continuum_remove_none_option(self):
+        self.continuum.items = [item for item in self.continuum.items
+                                if item['label'] != 'None']
+        self.continuum._apply_default_selection()
+
+    @property
+    def continuum_marks(self):
+        marks = {}
+        viewer = self.app.get_viewer(self._default_spectrum_viewer_reference_name)
+        if viewer is None:
+            return {}
+        for mark in viewer.figure.marks:
+            if isinstance(mark, LineAnalysisContinuum):
+                # NOTE: we don't use isinstance anymore because of nested inheritance
+                if mark.__class__.__name__ == 'LineAnalysisContinuumLeft':
+                    marks['left'] = mark
+                elif mark.__class__.__name__ == 'LineAnalysisContinuumCenter':
+                    marks['center'] = mark
+                elif mark.__class__.__name__ == 'LineAnalysisContinuumRight':
+                    marks['right'] = mark
+
+        if not len(marks):
+            if not viewer.state.reference_data:
+                # we don't have data yet for scales, defer initializing
+                return {}
+            # then haven't been initialized yet, so initialize with empty
+            # marks that will be populated once the first analysis is done.
+            marks = {'left': LineAnalysisContinuumLeft(viewer, visible=self.is_active),
+                     'center': LineAnalysisContinuumCenter(viewer, visible=self.is_active),
+                     'right': LineAnalysisContinuumRight(viewer, visible=self.is_active)}
+            shadows = [ShadowLine(mark, shadow_width=2) for mark in marks.values()]
+            # NOTE: += won't trigger the figure to notice new marks
+            viewer.figure.marks = viewer.figure.marks + shadows + list(marks.values())
+
+        return marks
+
+    def _update_continuum_marks(self, mark_x={}, mark_y={}):
+        for pos, mark in self.continuum_marks.items():
+            mark.update_xy(mark_x.get(pos, []), mark_y.get(pos, []))
+
+    def _get_continuum(self, dataset, spatial_subset, spectral_subset, update_marks=False):
+        if dataset.selected == '':
+            self._update_continuum_marks()
+            return None, None, None
+
+        if spatial_subset == 'per-pixel':
+            if self.app.config != 'cubeviz':
+                raise ValueError("per-pixel only supported for cubeviz")
+            if "_orig_spec" in self.dataset.selected_obj.meta:
+                full_spectrum = self.dataset.selected_obj.meta["_orig_spec"]
```

Comment:
@kecnry The bug I'm seeing is triggered by this line. If I use the `full_spectrum` from line 2059 in a notebook, I see a result. If I use this line, I see the traceback I mentioned here https://github.com/spacetelescope/jdaviz/pull/2587#issuecomment-1846099676 .

Reply:
curious, ok, I'll revert that logic and we can re-introduce it if we run into a case where its necessary.